Drop-In/src/androidTest/java/com/braintreepayments/api/AddCardFragmentUITest.kt

```diff
@@ -0,0 +1,137 @@
+package com.braintreepayments.api
+
+import androidx.fragment.app.testing.FragmentScenario
+import androidx.lifecycle.Lifecycle
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.action.ViewActions.click
+import androidx.test.espresso.action.ViewActions.typeText
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.matcher.ViewMatchers.*
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.braintreepayments.api.CardNumber.VISA
+import com.braintreepayments.api.dropin.R
+import com.braintreepayments.cardform.utils.CardType
+import junit.framework.TestCase.assertNull
+import org.hamcrest.CoreMatchers.not
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.util.concurrent.CountDownLatch
+
+@RunWith(AndroidJUnit4::class)
+class AddCardFragmentUITest {
+
+    private lateinit var countDownLatch: CountDownLatch
+
+    @Before
+    fun beforeEach() {
+        countDownLatch = CountDownLatch(1)
+    }
+
+    @Test
+    fun whenStateIsRESUMED_buttonTextIsAddCard() {
+        val scenario = FragmentScenario.launchInContainer(AddCardFragment::class.java, null, R.style.bt_drop_in_activity_theme)
+        scenario.moveToState(Lifecycle.State.RESUMED)
+
+        scenario.onFragment { fragment ->
+            fragment.dropInViewModel.setSupportedCardTypes(listOf(CardType.VISA))
+        }
+
+        onView(isRoot()).perform(waitFor(500))
+        onView(withId(R.id.bt_button)).check(matches(withText(R.string.bt_next)))
+    }
+
+    @Test
+    fun whenStateIsRESUMED_cardNumberFieldIsFocused() {
+        val scenario = FragmentScenario.launchInContainer(AddCardFragment::class.java, null, R.style.bt_drop_in_activity_theme)
+        scenario.moveToState(Lifecycle.State.RESUMED)
+
+        onView(withId(R.id.bt_card_form_card_number)).check(matches(isFocused()))
+    }
+
+    @Test
+    fun whenStateIsRESUMED_doesNotShowErrorForNonCardNumberError() {
+        val scenario = FragmentScenario.launchInContainer(AddCardFragment::class.java, null, R.style.bt_drop_in_activity_theme)
+        scenario.moveToState(Lifecycle.State.RESUMED)
+
+        scenario.onFragment { fragment ->
+            fragment.dropInViewModel.setSupportedCardTypes(listOf(CardType.VISA))
+            fragment.dropInViewModel.setCardTokenizationError(ErrorWithResponse.fromJson(Fixtures.CREDIT_CARD_EXPIRATION_ERROR_RESPONSE))
+
+            assertNull(fragment.cardForm.cardEditText.textInputLayoutParent?.error)
+        }
+    }
+
+    @Test
+    fun whenStateIsRESUMED_sendsAnalyticsEventToNotifyAddCardSelected() {
```

Comment:
```suggestion
    fun whenStateIsRESUMED_andAddCardSelected_sendsAnalyticsEvent() {
```
So that the "expected behavior" is the last part of the test name